pallets/hyperdrive/src/lib.rs

```diff
@@ -430,6 +442,11 @@ pub mod pallet {
             Self::process_action(payload_bytes)
         }
 
+        /// <HB SBP Milestone Review II
+        ///
+        /// You can remove the transactional macro since it's already happening out of the box.
+        ///
+        /// >
         #[transactional]
         fn process_action(message_bytes: &Vec<u8>) -> Result<(), ProcessMessageResult> {
```

Comment:
@hbulgarini Do you refer to the transaction started at the extrinsic level?

Our design requires the extrinsic `submit_message` calling `process_message` -> `process_action` **not to fail on errors (we hide errors after depositing an event at line 378)** happening during executing the message, but _selectively_ revert (partial) storage changes done during whatever executing involves.

I expanded the top-level `frame_support::pallet` macro both with and without the `#[transactional]` here and only with it I get the `process_action`'s function body wrapped with:

```rust
fn process_action(message_bytes: &Vec<u8>) -> Result<(), ProcessMessageResult> {
            use frame_support::storage::{with_transaction, TransactionOutcome};
            with_transaction(|| {
                let r = (|| {
                    {
                        // ...
                        T::ActionExecutor::execute(action) // <- whatever side-effects produced by this should be reverted on later failure
                            .map_err(|_| ProcessMessageResult::ActionFailed(raw_action.clone()))?;

                        Ok(())
                    }
                })();
                if r.is_ok() { TransactionOutcome::Commit(r) } else { TransactionOutcome::Rollback(r) }
            })
        }
```

Reply:
indeed but my understanding is that the `process_action` fn is going to be reverted anyway since it's being called from the extrinsic `submit_message` making all the storage commits revertible. 
But your comment is valid and now i doubt about my statement, maybe we might reproduce it in a test and see how it works? 

These two GH links might be useful as well:

https://github.com/paritytech/substrate/pull/11431
https://github.com/paritytech/substrate/issues/10806